src/simulation.rs

```diff
@@ -284,11 +286,12 @@ mod mocks {
                 .expect_set_simulated_timestep
                 .expect("set_simulated_timestep() was called unexpectedly");
 
-            assert_eq!(
-                expected_timestep, timestep,
+            assert!(
+                NearlyEq::eq(&expected_timestep, &timestep, &f64::eps()),
```

Comment:
I had to use `assert!()` here, because `assert_nearly_eq!()` doesn't support passing a custom panic message.